packaging/docker/Dockerfile

```diff
@@ -0,0 +1,388 @@
+# Dockerfile: compile and package Ultimate toolchains
+#------------------------------------------------------------------------------
+# Author: Manuel Bentele
+# Copyright (c) 2023
+#------------------------------------------------------------------------------
+# common build arguments for multi-staged Docker build
+#------------------------------------------------------------------------------
+ARG REPO_MAVEN_ROOT="trunk/source/BA_MavenParentUltimate"
+ARG REPO_FRONTEND_ROOT="trunk/source/WebsiteStatic"
+ARG REPO_DIR_BUILD="trunk/source/BA_SiteRepository/target/products"
+ARG REPO_DIR_TOOLS="releaseScripts/default/adds"
+ARG DIR_BUILD="/home/build"
+ARG DIR_ULTIMATE="${DIR_BUILD}/ultimate"
+ARG DIR_OUTPUT="${DIR_ULTIMATE}/${REPO_DIR_BUILD}"
+ARG DIR_OUTPUT_FRONTEND="${DIR_ULTIMATE}/${REPO_FRONTEND_ROOT}"
+ARG DIR_TOOLS="${DIR_ULTIMATE}/${REPO_DIR_TOOLS}"
+ARG USER_NAME="ultimate"
+ARG USER_GROUP="ultimate"
+ARG USER_HOME="/home/${USER_NAME}"
+
+#------------------------------------------------------------------------------
+# 1st build stage: compile and materialize Ultimate
+#------------------------------------------------------------------------------
+FROM alpine:3 AS build
+
+ARG REPO_URL="https://github.com/ultimate-pa/ultimate.git"
+ARG REPO_BRANCH="dev"
+
+ARG REPO_MAVEN_ROOT
+ARG REPO_DIR_BUILD
+ARG DIR_BUILD
+ARG DIR_ULTIMATE
+ARG DIR_OUTPUT
+
+# install build dependencies for an entire Ultimate build
+RUN apk add --no-cache "git" \
+                       "maven" \
+                       "openjdk21"
+
+# clone fresh Ultimate Git repository
+RUN mkdir -p "${DIR_BUILD}" && \
+    git clone --depth=1 --branch="${REPO_BRANCH}" "${REPO_URL}" "${DIR_ULTIMATE}"
+
+# build all Ultimate products
+WORKDIR "${DIR_ULTIMATE}"
+RUN mvn -f "${REPO_MAVEN_ROOT}" clean package -P materialize
+
+#------------------------------------------------------------------------------
+# 2nd build stage: compile and build the Ultimate web interface
+#------------------------------------------------------------------------------
+FROM build AS build-webfrontend
+
+ARG REPO_FRONTEND_ROOT
+ARG REPO_DIR_BUILD
+ARG DIR_BUILD
+ARG DIR_ULTIMATE
+ARG DIR_OUTPUT
+
+# install build dependencies for building the Ultimate web frontend
+RUN apk add --no-cache "gcc" \
+                       "g++" \
+                       "make" \
+                       "libc-dev" \
+                       "libc++-dev" \
+                       "linux-headers" \
+                       "python3" \
+                       "ruby-full" \
+                       "ruby-dev"
+
+# build the Ultimate web frontend
+WORKDIR "${REPO_FRONTEND_ROOT}"
+
+# patch the Jekyll installation to work under Ruby 3.4+
+RUN echo -e "gem 'csv'\n" \
+            "gem 'base64'\n" \
+            "gem 'bigdecimal'\n" \
+            "gem 'logger'" >> Gemfile
+
+ENV PATH "${DIR_OUTPUT}/CLI-E4/linux/gtk/x86_64:${PATH}"
+RUN python3 scripts/build.py
```

Comment:
The build cannot be configured by passing appropriate values, such as setting an arbitrary `baseurl` for Jekyll.
The used Python scripts must be adapted accordingly to make the Jekyll build configurable.

Reply:
There is a `--production` parameter that, among other things, sets the `baseurl`. But if you need more flexibility, then the script can be made more flexible, yes.